biorxiv/polka_subset_experiment/03_rerun_analyses_with_added_subset.py

```diff
@@ -0,0 +1,603 @@
+    )
+
+published_dates = published_dates.assign(
+    preprint_date=lambda x: pd.to_datetime(x.preprint_date.tolist()),
+    published_date=lambda x: pd.to_datetime(
+        x.published_date.apply(lambda y: y[0 : y.index(":")] if ":" in y else y)
+    ),
+).assign(time_to_published=lambda x: x.published_date - x.preprint_date)
+print(published_dates.shape)
+published_dates.head()
+# -
+
+polka_preprints_df = polka_preprints_df.assign(
+    biorxiv_base=lambda x: x.document.apply(lambda y: y.split("_")[0]),
+    version_count=lambda x: x.document.apply(
+        lambda y: int(y[y.index("v") + 1 :].split(".")[0])
+    ),
+)
+polka_preprints_df.head()
+
+polka_published_df = pd.read_csv("output/polka_et_al_pmcoa_embeddings.tsv", sep="\t")
+polka_published_df.head()
+
+polka_published_preprint_df = (
+    polka_et_al_mapped_df.drop(
+        ["Version", "MID", "IsCurrent", "IsLive", "ReleaseDate", "Msg"], axis=1
+    )
+    .assign(biorxiv_base=lambda x: x.biorxiv_doi.apply(lambda y: y.split("/")[1]))
+    .merge(
+        polka_preprints_df.drop([f"feat_{idx}" for idx in range(300)], axis=1),
+        on="biorxiv_base",
+    )
+    .drop("document", axis=1)
+    .merge(
+        published_dates.drop(["published_citation_count", "preprint_title"], axis=1),
+        on=["biorxiv_doi", "published_doi"],
+    )
+    .query(f"PMCID in {polka_published_df['document'].tolist()}")
+)
+polka_published_preprint_df.head()
+
+# +
+for col in ["preprint_date", "published_date"]:
+    polka_published_preprint_df[col] = pd.to_datetime(polka_published_preprint_df[col])
+
+polka_published_preprint_df["time_to_published"] = pd.to_timedelta(
+    polka_published_preprint_df["time_to_published"]
+)
+polka_published_preprint_df["days_to_published"] = polka_published_preprint_df[
+    "time_to_published"
+].dt.days
+print(polka_published_preprint_df.shape)
+polka_published_preprint_df.head()
+# -
+
+# ### Document version count plot
+
+biorxiv_published_distances = pd.read_csv(
+    "../publication_delay_experiment/output/preprint_published_distances.tsv", sep="\t"
+)
+biorxiv_published_distances["time_to_published"] = pd.to_timedelta(
+    biorxiv_published_distances["time_to_published"]
+)
+biorxiv_published_distances["days_to_published"] = biorxiv_published_distances[
+    "time_to_published"
+].dt.days
+biorxiv_published_distances.head()
+
+# +
+# Get smoothed linear regression line
+x = biorxiv_published_distances.version_count.values.tolist()
+
+y = biorxiv_published_distances.time_to_published.apply(
+    lambda x: x / timedelta(days=1)
+).tolist()
+
+xseq_2 = np.linspace(np.min(x), np.max(x), 80)
+
+results_2 = linregress(x, y)
+print(results_2)
+# -
+
+x_line = np.array(
+    [
+        biorxiv_published_distances["version_count"].min(),
+        biorxiv_published_distances["version_count"].max(),
+    ]
+)
+y_line = x_line * results_2.slope + results_2.intercept
+
+# Graph here?
+plt.figure(figsize=(8, 5))
+g = sns.violinplot(
```

Comment:
This makes sense. The longer it takes to publish the more preprint versions you would have. Is this serving as a positive control? I wonder if it would make sense to look at time to publish vs some other parameters?

Reply:
>This makes sense. The longer it takes to publish the more preprint versions you would have. Is this serving as a positive control? I wonder if it would make sense to look at time to publish vs some other parameters?

Not positive control. More like we wanted to show that preprints within their subset are being published at a faster rate than average. (Hence the red does below the regression line.)